miditoolkit/midi/containers.py

```diff
@@ -118,35 +113,34 @@ class TimeSignature(object):
 
     """
 
-    def __init__(self, numerator, denominator, time):
-        if not (isinstance(numerator, int) and numerator > 0):
-            raise ValueError(
-                f'{numerator} is not a valid `numerator` type or value')
-        if not (isinstance(denominator, int) and denominator > 0):
+    numerator: int
+    denominator: int
+    time: float
+
+    def __post_init__(self):
```

Comment:
Checks now done in `__post_init`
Type checks are not necessary as it is already done by `dataclass`